### PR TITLE
adds fileData method over api

### PR DESCRIPTION
**Pull request by andrewxhill**
quick attempt to include a fileData method in the library 

this doesn't currently work because of cross domain error against the gateway

### CODE REVIEW

**Comment by andrewxhill** [comment]
putting it here as an option... thinking through it though it probably makes sense at this point to include the fileData method in the core client api of go-textile to be exposed over the api

File: src/core/api.ts [hunk 1]
```diff
@@ -60,6 +61,25 @@ class API {
     }
     url.set('pathname', `/api/v${opts.version || 0}/`)
     this.baseURL = url.toString()
+
+    const gateway = new URL(this.opts.url)
```
Comment (andrewxhill) [inline src/core/api.ts:65]: no longer used here, but could be handy later

File: src/modules/files.ts [hunk 2]
```diff
@@ -113,4 +113,15 @@ export default class Files extends API {
...
+  /**
+   * Get raw data for a File
+   *
+   * @param file File object
+   * @returns Raw data
+   */
+  async fileData(hash: string): Promise<string> {
+    const response = await this.sendGet(`file/${hash}`)
+    return response.text() as Promise<string>
+  }
```
Comment (andrewxhill) [inline src/modules/files.ts]: Here is the method that would call this, https://github.com/textileio/go-textile/pull/714

working locally at least!

**Comment by andrewxhill** [comment]
updated with new api path

File: src/modules/files.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { API } from '../core/api'
-import { ApiOptions, Node, FilesList, Files as File, Keys, DirectoryList } from '../models'
+import { ApiOptions, Node, FilesList, Files as FilesType, File, Keys, DirectoryList } from '../models'
```
Comment (carsonfarmer) [inline src/modules/files.ts]: Ooo good call

File: src/modules/files.ts [hunk 4]
```diff
@@ -113,4 +113,15 @@ export default class Files extends API {
...
+
+  /**
+   * Get raw data for a File
+   *
+   * @param file File object
+   * @returns Raw data
+   */
+  async fileData(hash: string): Promise<string> {
+    const response = await this.sendGet(`file/${hash}`)
+    return response.text() as Promise<string>
```
Comment (carsonfarmer) [inline src/modules/files.ts]: if that endpoint returns raw bytes, we can get that here as well... might be more useful if you're doing things like grabbing raw data to convert to objects or similar?

Comment (carsonfarmer) [reply src/modules/files.ts]: We can get the response as an arrayBuffer or blob if we want: https://developer.mozilla.org/en-US/docs/Web/API/Response#Methods

**Comment by carsonfarmer** [comment]
Is this good to go now andrew? also, if we're moving to `file.add`, should we similarly change this to `file.data`? i'm a little behind on the changes in `go-textile`, so maybe this is already settled?

File: src/modules/files.ts (hunk 4 above)
Comment (andrewxhill) [reply src/modules/files.ts]: cool. that fetch lib suggested to use response.text() here so i just assumed to do it that but maybe they aren't mutually exclusive. i'm going to move this to a ticket, can you try to add? #

**Comment by andrewxhill** [comment]
cool, good call carson. updated. actually going to skip ahead of the core api and go with file.content as that's where they will move to

**Pull request merged**